base/src/org/adempiere/process/YearEndClosing.java

```diff
@@ -0,0 +1,183 @@
+		.list();		
+		for (MElementValue elementValue : accounts) 
+		{			
+
+			log.info (elementValue.getValue());		
+			balance = searchBalance(elementValue.getC_ElementValue_ID());
+			if (balance.equals(Env.ZERO))
+				continue;
+			createJournal(elementValue, journalBatch, acctSchema, balance);
+		}
+		return "@Created@: " + journalBatch.getDocumentNo();		
+	}	//	doIt
+	
+	private int createAccount(int elementValueId)
+	{   
+		
+		
+		MAccount acc = MAccount.get(getCtx(), getAD_Client_ID(),
+		getOrgId(), getAcctSchemaId(), 
+		elementValueId, 0,
+		0, 0, 0, 
+		0, 0, 0, 
+		0, 0, 0,
+		0, 0, 0,0,0,0, get_TrxName());
+		if (acc == null)
+			return 0;
+		return acc.get_ID();
+		
+	}
+	
+	private Boolean createJournal(MElementValue elementValue, MJournalBatch journalBatch, 
+			MAcctSchema acctSchema, BigDecimal balance)
+	{
+		MJournal journal = new MJournal(journalBatch);
+		int accountId = createAccount(elementValue.getC_ElementValue_ID());
+		log.info (elementValue.getValue());		
+		journal.setC_Currency_ID(acctSchema.getC_Currency_ID());
+		journal.setC_AcctSchema_ID(acctSchema.get_ID());
+		journal.setC_ConversionType_ID(MConversionType.getDefault(getAD_Client_ID()));
+		journal.setDescription(Msg.translate(Env.getCtx(), "Closing") + " " + elementValue.getValue() + " " + elementValue.getName());
+		journal.setDocumentNo(DB.getDocumentNo(getAD_Client_ID(), MJournal.Table_Name, get_TrxName()));
+		journal.setGL_Category_ID(getCategoryId());
+		journal.save();
+		if (!createLines(accountId, journal, elementValue, balance))
+			return false;				
+		return true;
+	}
+	
+	private Boolean createLines(int accountID, MJournal journal, 
+			MElementValue elementValue, BigDecimal balance)
+	{
+		String accounttype = elementValue.getAccountType();
+		MJournalLine debit = new MJournalLine(journal);
+		MJournalLine credit = new MJournalLine(journal);
+		if (balance.signum() == 0)
+			return true;
+		if (accounttype.equals(MElementValue.ACCOUNTTYPE_Revenue))
+		{
+			debit.setC_ValidCombination_ID(closingAccountId);
+			debit.setAmtSourceDr(balance.negate());
+			debit.save();
+			credit.setC_ValidCombination_ID(accountID);
+			credit.setAmtSourceCr(balance.negate());
+			credit.save();
+		}
+		else
+		{
+			debit.setC_ValidCombination_ID(closingAccountId);
+			debit.setAmtSourceDr(balance);
+			debit.save();
+			credit.setC_ValidCombination_ID(accountID);
+			credit.setAmtSourceCr(balance);
+			credit.save();
+		}
+		return true;
+	}
+	
+	private BigDecimal searchBalance(int elementValueId)
+	{
+	    ArrayList<Object> params = new ArrayList<Object>();	    
+	    params.add(MPeriod.getFirstInYear(getCtx(), getDateAcct(), 0).getStartDate()); 
+	    params.add(getDateAcct()); 
+	    params.add(elementValueId);
+		String sql = " SELECT COALESCE(sum(amtacctDr - amtacctCr), 0) FROM fact_acct f" +
+			" WHERE (dateacct BETWEEN ? AND ? )" +
+			" AND f.account_ID = ? ";
+		BigDecimal saldo = DB.getSQLValueBDEx(get_TrxName(), sql, params);
+		return saldo;

```

Comment:
@SusanneCalderon  Can you please use English names, there shouldn't be Spanish names in the code by convention?